packages/fluentui/react-builder/src/components/MessageComputer.tsx

```diff
@@ -0,0 +1,247 @@
+ * What about the "disabled" state?
+ */
+import SRMC from './SRMC-Definitions';
+import './SRMC-Rules-Win_JAWS';
+
+export default class MessageComputer {
+  private computedMessageParts: { [key: string]: string } = {
+    value: '',
+    name: '',
+    description: '',
+    type: '',
+    state: '',
+    position: '',
+    usage: '',
+  };
+
+  // Asynchronously computes and returns the complete screen reader message for the given element and platform.
+  async computeMessage(element: HTMLElement, platform: string): Promise<string> {
+    let definitionName = this.getDefinitionName(element, platform, 'stateRules');
+
+    // Determine and save the usage string
+    this.computedMessageParts.usage = '';
+    const usages = SRMC.usageStrings[platform][definitionName];
+    if (usages) {
+      // Begin if 1
+      if (usages['[default]']) {
+        // Begin if 2
+        this.computedMessageParts.usage = usages['[default]'];
+      } // End if 2
+      for (let usageName in usages) {
+        // Begin for 1
+        const split = usageName.split('=');
+        const state = split[0];
+        const stateValue = split[1];
+        const stateAndValueMatch = element.getAttribute(state) === stateValue;
+        const checkedDOMPropAndValueMatch =
+          state === 'checked' && (element as HTMLInputElement).checked.toString() === stateValue;
+        if (stateAndValueMatch || checkedDOMPropAndValueMatch) {
+          // Begin if 2
+          this.computedMessageParts.usage = usages[usageName];
+        } // End if 2
+      } // End for 1
+    } // End if 1
+
+    // compute the element's accessible description
+    let computedDescription = '';
+
+    // First, handle some special case conditions
+    let value;
+    if (definitionName === 'textarea') {
+      // Begin if 1
+      value = (element as HTMLTextAreaElement).value.trim();
+    } // End if 1
+    if (definitionName === 'textarea' && platform === 'Win/JAWS' && value) {
+      // Begin if 1
+      computedDescription = SRMC.stateStrings['Win/JAWS']['textarea']['[extra1]'];
+    } else {
+      // else if 1
+      const describedby = element.getAttribute('aria-describedby');
+      let descElement;
+      if (describedby) {
+        // Begin if 2
+        const computedDescriptionArr = [];
+        const describedbyArr = describedby.split(/\s+/);
+        for (let i = 0; i < describedbyArr.length; i++) {
+          // Begin for 1
+          descElement = document.getElementById(describedbyArr[i]);
+          if (descElement) {
+            // Begin if 3
+            computedDescriptionArr.push(descElement.textContent);
+          } // End if 3
+        } // End for 1
+        computedDescription = computedDescriptionArr.join('');
```

Comment:
this might be more readable:

```
computedDescription = describedby
  .split(/\s+/)
  .map(x => {
    return document.getElementById(describedbyArr[i]) ? descElement.textContent : nul
  })
  .filter(x => x !== null)
  .join('')
```

Reply:
Shouldn't it be rather like this:

```
computedDescription = element.getAttribute('aria-describedby')
    ?.split(/\s+/)
    .map(x => {
        return element.ownerDocument?.getElementById(x)?.textContent || null;
    })
    .filter(x => x !== null)
    .join('') || '';
```

Reply:
yeah something along those lines. Main point was that the `map` function might make your life easier here :)